airbyte-cdk/python/unit_tests/sources/streams/http/test_http.py

```diff
@@ -79,6 +97,20 @@ def test_stub_basic_read_http_stream_read_records(mocker):
     assert [{"data": 1}] == records
 
 
+def test_stub_basic_read_http_stream_read_records_as_messages(mocker, mock_datetime_now):
+    stream = StubBasicReadHttpStream()
+    blank_response = {}  # Send a blank response is fine as we ignore the response in `parse_response anyway.
+    mocker.patch.object(StubBasicReadHttpStream, "_send_request", return_value=blank_response)
```

Comment:
I feel like mocking private methods is a smell. I know it didn't originate in this PR, but it feels like it's telling us we need a dependency injection of the requester component

Reply:
+100
Would be nice to tackle https://github.com/airbytehq/airbyte/issues/3284. It would also help decouple [the simpleretriever from the HttpStream](https://github.com/airbytehq/airbyte/issues/14012)